contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1033,16 +1033,49 @@ update_GrantStmt(Node *n, const char *object, const char *obj_schema, const char
 	else if (obj_schema && stmt->objects)
 	{
 		RangeVar   *tmp = (RangeVar *) llast(stmt->objects);
-
 		tmp->schemaname = pstrdup(obj_schema);
 	}
 
 	if (grantee && stmt->grantees)
 	{
 		RoleSpec   *tmp = (RoleSpec *) llast(stmt->grantees);
-
+		if (strcmp(grantee, "public") == 0)
+		{
+			tmp->roletype = ROLESPEC_PUBLIC;
+		}
 		tmp->rolename = pstrdup(grantee);
 	}
+
+	if (priv && stmt->privileges)
+	{
+		AccessPriv *tmp = (AccessPriv *) llast(stmt->privileges);
+		tmp->priv_name = pstrdup(priv);
```

Comment:
1. stmt->privileges will always be non-NIL?
2. tmp->priv_name should be freed before allocating new memory

Reply:
1. Yes.
2. Are you suggesting this?
```
AccessPriv *tmp = (AccessPriv *) llast(stmt->privileges);
pfree(tmp->priv_name);
tmp->priv_name = pstrdup(priv);
```

Reply:
I verified and we don't need to pfree() this pointer explicitly since it's already NULL at this point.